### ISSUE TITLE
Improve an approximation quality of a PCE surrogate model

### ISSUE THREAD

**Issue by takafusui** [jonathf/chaospy#352]
Hi,

I have a stochastic model and interpolate model outcomes by PCE to quantity parametric uncertainty.
When a surrogate model is constructed on the arbitrary selected experimental design points (or training points), I estimate a generalization error based on the Eq. (27) in [the paper](https://sudret.ibk.ethz.ch/research/publications/preprints-archive/2016-007.html).

The number of experimental design points is 2 to 3 times {M + p \choose p}, where M is the number of uncertain parameters and p is the maximum order of polynomials, so I believe the number of experimental design points is fairy good.

My question is that are there ways to improve an approximation quality of a PCE surrogate model without increasing the number of model evaluations (experimental design points)?
(My model is stochastic and highly non-linear, and it requires non-negotiable time to simulate on GPU...)

For instance, especially in machine learning, we often standardize training data to have a mean 0 and a standard deviation 1. I am looking for this kind of technique that is common in UQ.

I appreciate any comments. Thank you in advance.

**Comment by jonathf** [comment]
What techniques and fine tuning gives the best result is a bit problem dependent. For example:
* What kind of values p and M are in your example?
* What probability distributions you are using?
* Are you using point collocation og pseudo spectral projection method?
* Are there any extra information the model can provide, like gradient, Hessian, known discontinuities, etc.?

PCE are not as sensitive to variation in input as ML, so there is less need for normalization, but it can be an issue if the scale is very different. E.g. `cp.J(cp.Uniform(0, 1e-5), cp.Uniform(0, 1e+5))` would likely be a problem. In that case I would consider using [generalized polynomial chaos](https://chaospy.readthedocs.io/en/master/user_guide/advanced_topics/generalized_polynomial_chaos.html) as a method for normalizing the input.

**Comment by takafusui** [comment]
Hi Jonathan,

thank you very much for your comments. I always start from a simple example, thus, so far I have:

- p=4, M=3
- Assume uniform distribution for each uncertain parameter
- Point collocation. I use `cp.fit_regression` then aim to minimize the mean square error (correct?). The number of experimental design points is 2 to 3 times {3 + 4 \choose 4}.
- The model outcomes should be continuous and have no kink from my intuitions.

The distributions of each uncertain parameter are in the same order so the problem should not be so complicated as you might expect...

Sorry I cannot paste any code here... I appreciate any comments as usual :)

**Comment by jonathf** [comment]
Yes, default `fit_regression` minimizes the mean square error.

Okay, when operating with such a low order and dimension count, and the fact that you have no "kinks", I would not be surprised if you got better results using [pseudo-spectral projection](https://chaospy.readthedocs.io/en/master/user_guide/main_usage/pseudo_spectral_projection.html).

You could try using full tensor rid, `rule="Gaussian"` and quadrature order == polynomial order.

**Comment by takafusui** [comment]
Dear Jonathan,

sorry for my late answer...
I have tried some options, including your suggested projection method.
I found the most effective option is to increase the degree of polynomial up to 9 with the regression method, and the generalization error satisfies the criteria in the literature.

If you do not have further comments from your side, I would like to close this issue. Thank you very much!
